File: typescript/src/vessel_sanctions_list_test.ts
import OnboardingBuddyClient from '@onboardingbuddy/onboardingbuddy-client'
import * as dotenv from 'dotenv'
dotenv.config()

const obAppKey = process.env.OB_APP_KEY || '';
const obApiKey = process.env.OB_API_KEY || '';
const obApiSecret = process.env.OB_API_SECRET || '';

const client = new OnboardingBuddyClient(obAppKey, obApiKey, obApiSecret);

const vesselSanctions = client.checkVesselSanctions({
  name: 'HWANG GUM SAN 2'
});

vesselSanctions
    .then(response => 
        { 
            console.log(`Matched: ${response.data.matched}`)
            if (response.data.match)
            {
                console.log(`Full Name: ${response.data.match[0].fullName}`)
                console.log(`Program: ${response.data.match[0].program}`)
                console.log(`Additional Info: ${response.data.match[0].additionalInfo}`)
                console.log(`Secondary Sanctions Risk: ${response.data.match[0].secondarySanctionsRisk}`)
                console.log(`Vessel Type: ${response.data.match[0].vesselType}`)
                console.log(`Vessel Flag: ${response.data.match[0].vesselFlag}`)
            }
        })
    .catch(error=> console.log(error.response.data));

File: python/vessel_sanctions_list_test.py
from onboarding_buddy_client.api.sanctions_api import SanctionsApi
from onboarding_buddy_client.models.vessel_sanctions_check_request_m import VesselSanctionsCheckRequestM
from onboarding_buddy_client import Configuration, ApiClient
from dotenv import load_dotenv
import os

load_dotenv()

# Configure the client with authentication headers
config = Configuration(
    api_key={
        "AppKey": os.getenv("OB_APP_KEY"),
        "ApiKey": os.getenv("OB_API_KEY"),
        "ApiSecret": os.getenv("OB_API_SECRET")
    }
)

# Initialize the API client
api_client = ApiClient(configuration=config)
sanctions_api = SanctionsApi(api_client)

# Submit the vessel sanction request
try:
    # Create a vessel sanction check request
    request = VesselSanctionsCheckRequestM(
        name="HWANG GUM SAN 2"
    )

    response = sanctions_api.vessel(vessel_sanctions_check_request_m=request)
    
    print(f"Matched: {response.matched}")
    print(f"Full Name: {response.match[0].full_name}")
    print(f"Program: {response.match[0].program}")
    print(f"Additional Info: {response.match[0].additional_info}")
    print(f"Secondary Sanctions Risk: {response.match[0].secondary_sanctions_risk}")
    print(f"Vessel Type: {response.match[0].vessel_type}")
    print(f"Vessel Flag: {response.match[0].vessel_flag}")
    
except Exception as e:
    print(f"Error: {e}")